client-v2/src/main/java/com/clickhouse/client/api/internal/HttpAPIClientHelper.java

```diff
@@ -580,4 +582,30 @@ public ClientException wrapException(String message, Exception cause) {
 
         return new ClientException(message, cause);
     }
+
+
+    /**
+     * Parses URL parameters.
+     * @param url
+     * @return Map of parameters
+     */
+    public static Map<String, String> parseUrlParameters(URL url) {
+        Map<String, String> params = new HashMap<>();
+
+        try {

```

Comment:
let's add a tests for parsing url with `custom_http_headers` and `custom_http_params`
see: https://clickhouse.com/docs/en/integrations/java/jdbc-driver#configuration